app/services/clients/base.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# Clients for interacting with external APIs
+module Clients
+  # Base class for harvesting clients
+  class Base
+    attr_reader :conn
+
+    def initialize(conn: nil)
+      @conn = conn || new_conn
+    end
+
+    def get_json(path:, params: {})
+      response = conn.get(path, params.compact)
+
+      raise Clients::Error, "Error: #{response.status}" unless response.success?
+
+      JSON.parse(response.body)
+    rescue Faraday::Error => e
+      raise Error, "Connection err: #{e.message}"
+    rescue JSON::ParserError => e
+      raise Error, "JSON parsing error: #{e.message}"
+    end
+
+    private
+
+    def new_conn
+      Faraday.new('example.com')
```

Comment:
Can this be `raise NotImplementedError`?

Reply:
Absolutely.

Reply:
Fixed